### PR TITLE
Multi Player Sitting (rework #122)

### PR DESCRIPTION
**Pull request by jonpas**
**When merged this pull request will:**
- Add everything in #122 in a better, more integrated way
- Add Arm Chair and fix inheritance on benches
- Improve variable naming

### CODE REVIEW

File: addons/sitting/functions/fnc_addSitActions.sqf [hunk 1]
```diff
@@ -17,26 +17,55 @@
...
-] call ACEFUNC(interact_menu,createAction);
-[_type, 0, ["ACE_MainActions"], _sitAction] call ACEFUNC(interact_menu,addActionToClass);
+private _sitPosition = getArray (_configFile >> QGVAR(sitPosition));
+private _interactPosition = getArray (_configFile >> QGVAR(interactPosition));
+
+if (count _sitPosition != count _interactPosition) exitWith {
+    WARNING_1("Invalid sitting configuration of %1!",_type);
+};
+
+if !((_sitPosition select 0) isEqualType []) then {
```
[the remark this answers is not in the document]
Comment (jonpas) [reply addons/sitting/functions/fnc_addSitActions.sqf]: I could, doesn't make a difference here though.

File: addons/sitting/functions/fnc_canSit.sqf [hunk 2]
```diff
@@ -20,6 +20,6 @@ params ["_seat", "_player"];
 
 // Sitting enabled, is seat object, not occupied and standing up (or not on a big slope)
 GVAR(enable) &&
-{isNil {_player getVariable QGVAR(isSitting)}} &&
-{!(_seat call ACEFUNC(common,owned))} &&
+{isNil {_player getVariable QGVAR(sittingStatus)}} &&
+{(_seat getVariable [QGVAR(seatsClaimed), []]) find true == -1} &&
```
Comment (Dystopian) [inline addons/sitting/functions/fnc_canSit.sqf]:
```suggestion
{
    private _seatsClaimed = _seat getVariable [QGVAR(seatsClaimed), []];
    _seatsClaimed isEqualTo []
    || {_seatsClaimed find false > -1}
} &&
```
I didn't test your code neither mine (will try later this day), but looks like your condition is `false` when any seat is claimed, so only one person would be able to multi-seat.

Comment (jonpas) [reply addons/sitting/functions/fnc_canSit.sqf]: Good point, I should just be checking that specific index (seat pos) or empty!

**Comment by jonpas** [comment]
@Dystopian if you can give it a test that'd be great. I tested in editor with 2 units.

File: addons/sitting/functions/fnc_sit.sqf [hunk 3]
```diff
@@ -1,22 +1,24 @@
...
  * 0: Seat <OBJECT>
  * 1: Player <OBJECT>
+ * 2: Seat Position <NUMBER> (default: 0)
  *
  * Return Value:
  * None
  *
  * Example:
- * [seat, player] call acex_sitting_fnc_sit;
+ * [seat, player] call acex_sitting_fnc_sit
```
Comment (Dystopian) [inline addons/sitting/functions/fnc_sit.sqf]: `cursorObject` instead of `seat` in all functions examples would be much more useful

**Comment by Dystopian** [comment]
tested, works perfectly (SP)

File: addons/sitting/functions/fnc_sit.sqf (hunk 3 above)
Comment (jonpas) [reply addons/sitting/functions/fnc_sit.sqf]: It's an object, that's all that matters, you can get it in thousands of different ways. :)

Comment (Dystopian) [reply addons/sitting/functions/fnc_sit.sqf]: I meant for copy-paste, just paste to debug console and hit `EXEC`

Comment (PabstMirror) [reply addons/sitting/functions/fnc_sit.sqf]: having code example you can use instantly is always nice

Comment (jonpas) [reply addons/sitting/functions/fnc_sit.sqf]: Good point, I am always too lazy with copy/pasting. :laughing:

**Comment by jonpas** [comment]
@PabstMirror how? MP or SP? Which bench?

**Comment by jonpas** [comment]
Oh but for AI I won't add special checks, it's up to whoever calls it to also call canSeat. Or not, people find interesting use cases. :D

**Comment by jonpas** [comment]
> I suppose it can be done in MP when some players try to seat to the same seat simultaneously.

Almost missed that, that's probably a good reason to add it.

**Comment by PabstMirror** [comment]
seatPos array was only getting set to true on first run `if (_seatsClaimed isEqualTo []) then {`
seems to work fine now

**Comment by jonpas** [comment]
Wow, I should not work on this stuff at 3am... I now remember thinking I have to add normal handling and not just first run. :D

**Pull request merged**